spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/channel/JacksonChannelMessageStorePreparedStatementSetter.java

```diff
@@ -0,0 +1,183 @@
+ * and embeds class type information for secure deserialization.
+ * <p>
+ * <b>Usage Example:</b>
+ * <pre>{@code
+ * &#64;Bean
+ * JdbcChannelMessageStore messageStore(DataSource dataSource) {
+ *     JdbcChannelMessageStore store = new JdbcChannelMessageStore(dataSource);
+ *     store.setChannelMessageStoreQueryProvider(new PostgresChannelMessageStoreQueryProvider());
+ *
+ *     // Enable JSON serialization
+ *     store.setPreparedStatementSetter(
+ *         new JacksonChannelMessageStorePreparedStatementSetter("com.example"));
+ *     store.setMessageRowMapper(
+ *         new JacksonMessageRowMapper("com.example"));
+ *
+ *     return store;
+ * }
+ * }</pre>
+ * <p>
+ * <b>Database Column Type:</b>
+ * This implementation requires a text-based column type that supports JSON storage:
+ * <ul>
+ *     <li>PostgreSQL: {@code JSONB} (recommended) or {@code JSON}</li>
+ *     <li>MySQL: {@code JSON} </li>
+ *     <li>H2: {@code CLOB} </li>
+ * </ul>
+ * <p>
+ * <b>Note:</b> The standard Spring Integration schemas use {@code BLOB}/{@code BYTEA} columns
+ * which are not suitable for human-readable JSON storage. Use the provided JSON-specific
+ * schemas from {@code schema-*-json.sql} files.
+ *
+ * @author Yoobin Yoon
+ * @since 7.0
+ *
+ * @see JacksonMessageRowMapper
+ * @see JacksonMessagingUtils#messagingAwareMapper(String...)
+ */
+public class JacksonChannelMessageStorePreparedStatementSetter extends ChannelMessageStorePreparedStatementSetter {
+
+	private final ObjectMapper objectMapper;
+
+	/**
+	 * Create a new {@link JacksonChannelMessageStorePreparedStatementSetter} with the
+	 * default trusted packages from {@link JacksonMessagingUtils#DEFAULT_TRUSTED_PACKAGES}.
+	 * <p>
+	 * This constructor is suitable when you only need to serialize/deserialize
+	 * standard Spring Integration and Java classes.
+	 */
+	public JacksonChannelMessageStorePreparedStatementSetter() {
+		this(new String[0]);
+	}
+
+	/**
+	 * Create a new {@link JacksonChannelMessageStorePreparedStatementSetter} with
+	 * additional trusted packages for deserialization.
+	 * <p>
+	 * The provided packages are appended to the default allow-list from
+	 * {@link JacksonMessagingUtils#DEFAULT_TRUSTED_PACKAGES}, enabling deserialization
+	 * of custom payload types.
+	 * <p>
+	 * <b>Package Matching:</b>
+	 * <ul>
+	 *   <li>{@code "com.example"} – trusts {@code com.example} and all subpackages
+	 *       (e.g., {@code com.example.MyClass}, {@code com.example.sub.AnotherClass})</li>
+	 *   <li>{@code "com.example.MyClass"} – trusts only the specific class</li>
+	 *   <li>{@code "*"} – trusts all packages (not recommended for production)</li>
+	 * </ul>
+	 * <p>
+	 * <b>Note:</b> Subpackages are automatically included without wildcards.
+	 *
+	 * @param trustedPackages the additional packages to trust for deserialization
+	 */
+	public JacksonChannelMessageStorePreparedStatementSetter(String... trustedPackages) {
```

Comment:
This is something what is not used on serialization. Therefore this option is redundant.